test/RPCSession.cpp

```diff
@@ -296,40 +296,7 @@ void RPCSession::test_mineBlocks(int _number)
 {
 	u256 startBlock = fromBigEndian<u256>(fromHex(rpcCall("eth_blockNumber").asString()));
 	BOOST_REQUIRE(rpcCall("test_mineBlocks", { to_string(_number) }, true) == true);
-
-	// We auto-calibrate the time it takes to mine the transaction.
-	// It would be better to go without polling, but that would probably need a change to the test client
-
-	auto startTime = std::chrono::steady_clock::now();
-	unsigned sleepTime = m_sleepTime;
-	size_t tries = 0;
-	for (; ; ++tries)
-	{
-		std::this_thread::sleep_for(chrono::milliseconds(sleepTime));
-		auto endTime = std::chrono::steady_clock::now();
-		unsigned timeSpent = std::chrono::duration_cast<std::chrono::milliseconds>(endTime - startTime).count();
-		if (timeSpent > m_maxMiningTime)
-			BOOST_FAIL("Error in test_mineBlocks: block mining timeout!");
-		if (fromBigEndian<u256>(fromHex(rpcCall("eth_blockNumber").asString())) >= startBlock + _number)
-			break;
-		else
-			sleepTime *= 2;
-	}
-	if (tries > 1)
-	{
-		m_successfulMineRuns = 0;
-		m_sleepTime += 2;
-	}
-	else if (tries == 1)
-	{
-		m_successfulMineRuns++;
-		if (m_successfulMineRuns > 5)
-		{
-			m_successfulMineRuns = 0;
-			if (m_sleepTime > 2)
-				m_sleepTime--;
-		}
-	}
+	BOOST_REQUIRE(fromBigEndian<u256>(fromHex(rpcCall("eth_blockNumber").asString())) == startBlock + _number);
```

Comment:
I was told ``test_mineBlocks`` is always synchronous now and we don't need to poll anymore.

Reply:
Oh wow, this is fantastic news! Did you measure whether it has an impact on running time?

Reply:
I see ~45% speedup for running the end-to-end tests only - but that's only one run and not particularly in a "controlled environment" - but I think we can expect some improvement here. I'm much more worried about the changes in the semantics tests below, though - it would be good to have an explanation for them.

Reply:
But the difference is just a block number offset, right? Honestly, I don't care too much about which data we get as long as we can tell from that data that the correct opcode is used by the code generator.

Reply:
Ok - I'm not sure: ``block.number`` now seems to return the number of the *previous* block, not the number of the *current* block, when I compare to the block numbers reported by ``aleth``. I would feel better if I knew why that is :-). But yes, it's probably not be a "real" issue, but rather something about our test setup or ``aleth``'s test mode - so it should be safe to ignore, although it's strange.